cfn-resources/global-cluster-config/cmd/resource/resource.go

```
@@ -0,0 +1,376 @@
+			HandlerErrorCode: cloudformation.HandlerErrorCodeInvalidRequest}, nil
+	}
+
+	if len(remove) > 0 {
+		if err := removeManagedNamespaces(context.Background(), client, remove, projectID, clusterName); err != nil {
+			_, _ = logger.Debugln("Error -Unable to remove Namespace")
+		}
+	}
+	if currentModel.RemoveAllZoneMapping != nil && *currentModel.RemoveAllZoneMapping {
+		_, _, err := client.GlobalClusters.DeleteCustomZoneMappings(context.Background(), projectID, clusterName)
+		if err != nil {
+			return progressevents.GetFailedEventByCode(fmt.Sprintf("Failed to custom zones : %s", err.Error()),
+				cloudformation.HandlerErrorCodeInvalidRequest), nil
+		}
+	}
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Delete Complete",
+	}, nil
+}
+
+func setup() {
+	util.SetupLogger("mongodb-atlas-global-cluster-config")
+}
+
+func isExist(client mongodbatlas.Client, currentModel *Model) bool {
+	setup()
+	config, _, err := ReadConfig(client, currentModel)
+	_, _ = logger.Debugf("%v", config)
+	if err == nil && config != nil {
+		return true
+	}
+	return false
+}
+
+// function to validate inputs to all actions
+func validateModel(fields []string, model *Model) *handler.ProgressEvent {
+	return validator.ValidateModel(fields, model)
+}
+
+func removeManagedNamespaces(ctx context.Context, conn *mongodbatlas.Client, remove []ManagedNamespace, projectID, clusterName string) error {
+	for _, m := range remove {
+		addManagedNamespace := &mongodbatlas.ManagedNamespace{
+			Collection:     cast.ToString(m.Collection),
+			Db:             cast.ToString(m.Db),
+			CustomShardKey: cast.ToString(m.CustomShardKey),
+		}
+		addManagedNamespace.IsCustomShardKeyHashed = pointy.Bool(*m.IsCustomShardKeyHashed)
+		addManagedNamespace.IsShardKeyUnique = pointy.Bool(*m.IsShardKeyUnique)
+		_, _, err := conn.GlobalClusters.DeleteManagedNamespace(ctx, projectID, clusterName, addManagedNamespace)
+
+		if err != nil {
```

should we return this error?